app/routes/events/components/Admin.tsx

```diff
@@ -21,51 +23,47 @@ const DeleteButton = ({ eventId, title }: ButtonProps) => {
   const dispatch = useAppDispatch();
   const navigate = useNavigate();
 
-  let deleteEventButton;
-
-  if (eventName === title) {
-    deleteEventButton = (
-      <ConfirmModal
-        title="Slett arrangement"
-        message="Er du sikker på at du vil slette dette arrangementet?"
-        onConfirm={() =>
-          dispatch(deleteEvent(eventId)).then(() => {
-            navigate('/events');
-          })
-        }
-      >
-        {({ openConfirmModal }) => (
-          <Button onClick={openConfirmModal} danger>
-            <Icon name="trash" size={19} />
-            Slett
-          </Button>
-        )}
-      </ConfirmModal>
-    );
-  }
-
   return (
-    <div>
-      {show === false && (
+    <>
+      {!show ? (
         <Button danger onClick={() => setShow(true)}>
           <Icon name="trash" size={19} />
           Slett arrangement
         </Button>
-      )}
-      {show && (
+      ) : (
         <>
-          <span>Skriv inn navnet på arrangementet du vil slette:</span>
-          <input
+          <label htmlFor="delete-event">
+            Skriv inn navnet på arrangementet du vil slette
+          </label>
+          <TextInput
+            id="delete-event"
             type="text"
-            id="slettArrangement"
+            prefix="warning"
             placeholder="Arrangementnavn"
             onChange={(e) => setEventName(e.target.value)}
-          />{' '}
-          <br />
-          {deleteEventButton}
+          />
+
+          {eventName === title && (
+            <ConfirmModal
+              title="Slett arrangement"
+              message="Er du helt sikker på at du vil slette dette arrangementet?"
+              onConfirm={() =>
+                dispatch(deleteEvent(eventId)).then(() => {
+                  navigate('/events');
+                })
+              }
+            >
+              {({ openConfirmModal }) => (
+                <Button onClick={openConfirmModal} danger>
+                  <Icon name="trash" size={19} />
+                  Slett
+                </Button>
+              )}
```

Comment:
I see that this an arrow key function. But for love of all that is holy, I can't understand how the first line is supposed to work. Could you clarify?

Reply:
It's nice that you're asking - I agree that this is kind of weird javascript.

The `ConfirmModal` takes in what we call "children" as a prop. React let's us pass this prop into the component by wrapping the `ConfirmModal` around the "children", which in this case is the delete button. This makes for much better readability, and you can see another example of this pattern [here](https://reactpatterns.js.org/docs/function-as-child-component/).

Now, what's happening on the line that you point out. Our `ConfirmModal` takes the `children` prop and puts it within itself, but in addition it also [passes a `openConfirmModal` function as a prop to its `children`](https://github.com/webkom/lego-webapp/blob/master/packages/lego-bricks/src/components/Modal/ConfirmModal.tsx#L118). Kind of weird, right? This means that the children of `ConfirmModal` (here, the code you see above) has access to this function `openConfirmModal`. But why the weird brackets and parentheses? Well, what we're doing here is ["destructing"](https://dev.to/angski/destructuring-props-in-react-4hfc) the function out of the props. This is sort of hacky javascript, as it lets us extract (destruct) things from objects and assign them to variables (in this case, `openConfirmModal`).

Does that make kinda sense? Good question!